examples/Linux/sentry.c

```diff
@@ -0,0 +1,324 @@
+            fflush(stdout);
+        }
+
+        /* Send the read data into the JSON parser */
+        ret = wolfsentry_config_json_feed(jps, buf, n, err_buf, sizeof err_buf);
+        if (ret < 0) {
+            fprintf(stderr, "%.*s\n", (int)sizeof err_buf, err_buf);
+            printf("error while loading wolfSentry config file");
+            fflush(stdout);
+        }
+        if ((n < sizeof buf) && feof(f))
+            break;
+    }
+    fclose(f);
+
+    /* Clean up the JSON parser */
+    if ((ret = wolfsentry_config_json_fini(&jps, err_buf, sizeof err_buf)) < 0) {
+        fprintf(stderr, "%.*s\n", (int)sizeof err_buf, err_buf);
+        printf("error while loading wolfSentry config file");
+        fflush(stdout);
+    }
+
+    return 0;
+}
+
+/* Check a TCP connection with wolfSentry. This is called for connect and
+ * disconnect so wolfSentry can count the simultaneous connections */
+int sentry_action(struct tcp_pcb *pcb, sentry_action_type action)
+{
+    wolfsentry_errcode_t ret;
+    wolfsentry_action_res_t action_results;
+    /* Note that sa.addr is 0 bytes, addr_buf essentially enlarges this to the correct size */
+    struct {
+        struct wolfsentry_sockaddr sa;
+        byte addr_buf[4];
+    } remote, local;
+    u32_t remoteip = pcb->remote_ip.addr;
+    u32_t localip = pcb->local_ip.addr;
+
+    /* Connect will increment the connection count in wolfSentry, disconnect
+     * will decrement it */
+    switch(action) {
+        case SENTRY_ACTION_CONNECT:
+            action_results = WOLFSENTRY_ACTION_RES_CONNECT;
+            break;
+        case SENTRY_ACTION_DISCONNECT:
+            action_results = WOLFSENTRY_ACTION_RES_DISCONNECT;
+            break;
+        case SENTRY_ACTION_NONE:
+        default:
+            break;
+    }
+
+    /* Setup sockaddr information to send to wolfSentry */
+    remote.sa.sa_family = WOLFSENTRY_AF_INET;
+    remote.sa.sa_proto = IPPROTO_TCP;
+    remote.sa.sa_port = pcb->remote_port;
+    /* Essentially a prefix size, wolfSentry uses the lesser of this and the
+     * rule in JSON as to how much of the IP address to compare */
+    remote.sa.addr_len = 32; // prefix size
+    remote.sa.interface = 0;
+    memcpy(remote.sa.addr, &remoteip, 4);
+
+    local.sa.sa_family = WOLFSENTRY_AF_INET;
+    local.sa.sa_proto = IPPROTO_TCP;
+    local.sa.sa_port = pcb->local_port;
+    local.sa.addr_len = 32;
+    local.sa.interface = 0;
+    memcpy(local.sa.addr, &localip, 4);
+
+    /* Send the details of this to wolfSentry and get the result */
+    ret = wolfsentry_route_event_dispatch_with_inited_result(
+            wolfsentry,
+            &remote.sa,
+            &local.sa,
+            WOLFSENTRY_ROUTE_FLAG_DIRECTION_IN,
+            "call-in-from-echo",
+            strlen("call-in-from-echo"),
+            NULL,
+            NULL,
+            NULL,
+            &action_results);
+
+    printf("TCP Sentry return: %d\n", ret);
```

Comment:
should use `WOLFSENTRY_ERROR_FMT` and `WOLFSENTRY_ERROR_FMT_ARGS()` to make these return codes legible. Note they are decodable+meaningful whether positive or negative.